lib/babe/verify.go

```diff
@@ -109,6 +110,23 @@ func (b *Session) verifyAuthorshipRight(slot uint64, header *types.Header) (bool
 		return false, fmt.Errorf("could not verify signature")
 	}
 
-	// TODO: check if the producer has equivocated, ie. have they produced a conflicting block?
+	// #553 check if the producer has equivocated, ie. have they produced a conflicting block?
+	hashes, err := b.blockState.GetAllHashesForParentDepth(header)
+	if err != nil {
+		return false, err
+	}
+
+	for key, v := range hashes {
+		//get header
+		h, err := b.blockState.GetHeader(key)
+		if err != nil {
+			return false, fmt.Errorf("could not verify digests, key %s, depth, %d", key, v)
+		}
+		//compare digest
+		if reflect.DeepEqual(h.Digest, header.Digest) {
+			return false, fmt.Errorf("duplicated digest")
```

Comment:
don't need to return an error, just `false`

Reply:
well, if I dont return a error how can I test that it happened?